### ISSUE TITLE
Views with sub-regions not handling views properly

### ISSUE THREAD

**Issue by jonmassot** [marionettejs/backbone.marionette#3612]
### Description

 1. I am not really sure this is a bug or not, but I will do my best to explain what I encountered. So, I have a generic `View` that add a `CollectionView` that is used to show some collections in different places. Let's call that view `Jobs`. In my app, I have the section where I manage jobs using the view. But I also have a `Resources` view where I can add/delete jobs associated to that resource. The resource page uses sub-regions, since I want to be able to manage the resource info and the jobs associated to him. 

```
main_region -> Jobs
main_region -> Resources -> Resource -> Info (sub-region)
                                                                -> Jobs (sub-region)
```
The thing is, whenever I switch from the `Resource` view to the `Jobs` view, I get an error that looks like `An "el" must exist in DOM for this region mnr388`. After debugging, I finally figured the views into sub-regions are not destroy, meaning `onDestroy` is not triggered, while the main view `onDestroy` is called.

 2. I fixed this temporarily with:
```
const region = this.getRegion('main_panel');
if (region.hasView()) {
    region.empty();
}
``` 
 3. Not the best, but I'm not sure if it's the right approach

### Expected behavior

`onDestroy` to be called in views within sub-regions

### Actual behavior

`onDestroy` not called in views within sub-regions

### Environment

1. Marionette version: 3.5.1
2. Backbone version: 1.3.3
3. Additional build tools, etc: No build tools involved at the development step, beside SASS with grunt

**Comment by paulfalgout** [comment]
I'll probably need to see a jsfiddle or something.  I can't reproduce the issue: https://jsfiddle.net/67xq2m3n/

**Comment by jonmassot** [comment]
@paulfalgout I forked your example so that I try to repro the same behavior I have:
https://jsfiddle.net/qchLa60w/8/

It might be my implementation that is the problem

**Comment by paulfalgout** [comment]
Your example works.. there's just a type in the fiddle `  template: _.template('-> div id="job"><%- depth %><</div>'),` should be `'-> <div id="job"...`  missing the opening `<`

**Comment by jonmassot** [comment]
Yeah, I figured that much and updated the fiddle to come as close as possible as to what I have and it seems to work fine.

**Comment by paulfalgout** [comment]
I am curious as to what is causing this.. I'm just not sure where to start looking

**Comment by jonmassot** [comment]
Same here. When camping the actual `this.empty()` in the view, it's the only way to not have the error.

**Comment by paulfalgout** [comment]
Maybe this'll help.. there are 2 scenarios in which this error can occur.  1 when showing a view in a region.. if show occurs but the region can find the `el` it'll error..  _or_ and I'd bet this is the one biting you.. if the region is being emptied, but it doesn't currently have a view, it'll try and remove the DOM from the region's `el`, but it'll error if it can't find the `el`.  It seems to me there might be an extra `empty` maybe?  after the view is already destroyed perhaps?  Or there's a region that never matched a selector in the template?

**Comment by jonmassot** [comment]
The thing is that I don't get the error with this code:
```
            const region = this.getRegion('main_panel');
            if (region.hasView()) {
                region.empty();
            }
```
If I remove that code, I get the error.

**Comment by paulfalgout** [comment]
and if you remove the `region.hasView` check?

**Comment by paulfalgout** [comment]
and if you only remove the `if (region.hasView()) {` check?

**Comment by jonmassot** [comment]
If I leave the `.empty`, I don't get the error. But if I remove the `.empty`, I do get the error

**Comment by paulfalgout** [comment]
how about?
```js
console.log(this.isRendered());  //what's this
const region = this.getRegion('main_panel');
console.log(region.hasView()); //what's this
region.empty();
```

**Comment by jonmassot** [comment]
When I load `resource`:
```
true
false
```
and when I load jobs right after:
```
true
true
```

**Comment by paulfalgout** [comment]
hmm ok.  really not sure at this point.

**Comment by jonmassot** [comment]
and that's the closest I can get with jsfiddle: https://jsfiddle.net/qchLa60w/35/

No worries, thanks a lot for the help! I know now that it should work and there's something on my part that is mixing things up!
